models/src/main/scala/coop/rchain/models/HashM.scala

```diff
@@ -62,7 +62,9 @@ object HashM extends HashMDerivation {
   implicit val BitSetHash: HashM[BitSet]                     = opaqueHash
   implicit val ByteStringHash: HashM[ByteString]             = opaqueHash
   implicit val Blake2b512RandomHash: HashM[Blake2b512Random] = opaqueHash
-  implicit def alwaysEqualHash[A]: HashM[AlwaysEqual[A]]     = opaqueHash
+  // This should be okay since injections are not recursive
+  implicit val InjectionsMapHash: HashM[Map[String, Par]] = opaqueHash
```

Comment:
@ArturGajowy would be nice if you could confirm that this is actually okay.

Reply:
Well... that implicit is going to be used whenever you use a `Map[String, Par]` in any of our datatypes we use HashM for.

It'd be safer to derive a HashM instance:
`implicit def mapHash[A: HashM, B: HashM]: HashM[Map[A, B]]`

Reply:
Do you have an idea on how to do that in a stack-safe manner? I'm a little lost here.

Reply:
I guess alternatively I could move `NormalizerEnv` to the models module and do `implicit val InjectionsMapHash: HashM[NormalizerEnv] = opaqueHash`.

Reply:
Yeah, that could work. I'd place the implicit on NormalizerEnv companion object

Reply:
Frankly, it'll likely work just as it is know - we're simply trying to reduce surface area for bugs to sneak in